### PR TITLE
Trainee finish task

### PR DESCRIPTION
**Pull request by lethihaivan**
Screenshot from 2021-04-15 16-20-52
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/35014)

## WHAT
-As a trainee, I can receive a task then I can finish a task

## Evidence (Screenshot or Video)
Screenshot from 2021-04-15 16-22-17

### CODE REVIEW

**Comment by lethihaivan** [comment]
ready

File: app/controllers/trainee/course_subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Trainee::CourseSubjectsController < UsersController
+  before_action :load_course_subject, :load_user_subject,
+                :load_user_tasks, :load_subject, :load_tasks, only: :show
+  def show; end
```
Comment (vanvtt-0952) [inline app/controllers/trainee/course_subjects_controller.rb:5]: line trống trc line này

File: app/controllers/trainee/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    flash[:warning] = t("user_subjects.task.not_found")
+    redirect_to trainee_course_subject_path @course_subject
+  end
+
+  def correct_user
+    redirect_to root_url unless current_user? @user_subject.user
+  end
+
+  def load_task
+    @task = Task.find_by id: params[:id]
```
Comment (vanvtt-0952) [inline app/controllers/trainee/tasks_controller.rb:50]: handle exception ở đây đi em

File: app/controllers/trainee/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  end
+
+  def load_user_task
+    @user_task = UserTask.find_by user_subject_id: @user_subject.id,
```
Comment (vanvtt-0952) [inline app/controllers/trainee/tasks_controller.rb:58]: handle exception ở đây đi em

**Comment by lethihaivan** [comment]
ready

**Pull request merged**